src/transformState.js

```diff
@@ -5,7 +5,25 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const char of actions) {
+    if (char['type'] === 'addProperties') {
+      for (const keys in char['extraData']) {
```

Comment:
It's better to use `.` notation to get access to object properties. Usually we use square brackets notation when we don't know the exact name of the property
```suggestion
      for (const keys in char.extraData) {
```